[Zen Lee] iP

zenlyj opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

zenlyj committed:
Implement Level-1

zenlyj committed:
Add java docs for Level-1

zenlyj committed:
Implement Level-2

zenlyj committed:
Implement Level-3

zenlyj committed:
Implement Level-4

zenlyj committed:
Fix ToDo adding bug and stop 'Task' adding

zenlyj committed:
Implement Automated Text UI Testing

zenlyj committed:
Rename Duke to Jarvis

zenlyj committed:
Change Duke to Jarvin

zenlyj committed:
Change Duke to Jarvis

zenlyj committed:
Merge branch 'master' of https://github.com/zenlyj/ip

zenlyj committed:
Add invalid command and description exceptions

zenlyj committed:
Add more exceptions to handle invalid inputs

zenlyj committed:
Implement delete function

zenlyj committed:
Implement Task Storage and Task Retrieval

Keeps track of all tasks with tasks.txt
User is able to retrieve previously added tasks even after restarting
the application

zenlyj committed:
Implement date and time in tasks

Allows bot to read & display date and time for deadlines and events
Bot reads dates in format: "yyyy-mm-dd" and time in 24hr format
Bot displays dates in format: "MONTH dd yyyy" and time in 12hr format

zenlyj committed:
Merge branch 'branch-Level-7'

Adding file storage implementations

zenlyj committed:
Merge branch 'branch-Level-8'

Allow program to read & display date and time for events and deadlines

zenlyj committed:
Create User Interface Class

Abstraction for printing and reading operations

zenlyj committed:
Create TaskList class

Representation of list of user's tasks
Allows for addition, removal of tasks as well as retrieval of all tasks

zenlyj committed:
Create Parser class and Command classes

Parser class
* Parses user input and creates command objects
* Checks for invalid inputs and displays error messages

Command classes
* Represents various commands of the bot
* After parsing user input, commands are executed to perform necessary
  operations

zenlyj committed:
Remove unnecessary exception classes

zenlyj committed:
Organize into packages

Create duke, command, task, exception directories

zenlyj committed:
Remove method that returns ArrayList

Encapsulates the ArrayList in the TaskList class by
* Creating listIterator method for iterating over all tasks
* Creating getTask method to return individual tasks

zenlyj committed:
Rename getDetails to getDateTime

For better description of what the method returns

zenlyj committed:
Make task storage receive a path from Duke

Modularity of TaskStorage class

zenlyj committed:
Add simple JUnit tests

Tests
* Check if add command adds task to task list
* Check if deadline can be marked done/undone
* Check if deadline description is returned correctly
* Check if deadline date and time is returned in correct
  String format

zenlyj committed:
Change fields to final and FileWriter to local var

Ui and DateTimeFormat should be delcared final
FileWriter is only used locally in the storeData method

zenlyj committed:
Implement Find function

Allows user to search for a task using a keyword

zenlyj committed:
Add java docs to all classes and methods

Described classes and methods with comments

zenlyj committed:
Adhere to coding standard

Changed indentation of switch-case statements
Changed name of setter methods
Changed class variable from public to private

zenlyj committed:
Add spacing after header comments

Previously, there was no spacing between description and parameter
section

zenlyj committed:
Merge branch 'Level-9'

Implemented Find command

zenlyj committed:
Merge branch 'branch-A-CodingStandard'

Adhere to the coding standard

zenlyj committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDocs to all classes and methods

zenlyj committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

zenlyj committed:
Set up gradle to run Duke and unit tests

zenlyj committed:
Remove old test directory

Shifted test directory to support gradle unit testing

zenlyj committed:
Merge branch 'branch-A-Gradle'

zenlyj committed:
Add GUI to Duke

Added a simple GUI to Duke using JavaFX

zenlyj committed:
Set gradle to run Duke GUI

Edited main class in build.gradle to point to GUI launcher

zenlyj committed:
Edit TaskList test

Set test to fail if an unexpected exception is thrown

zenlyj committed:
Update Java Docs

Add Java Docs to new methods as well as old methods that were modified

zenlyj committed:
Merge branch 'branch-Level-10'

Implement GUI for Duke and update Java Docs

Cp-John commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,143 @@
+        String command = split[0];
+        Command res;
+        try {
+            switch (command) {
```
Perhaps abstracting part of the code into methods with reasonable names is better in terms of readability?

Cp-John commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,143 @@
+            throw new DukeArgumentException("You have not entered a keyword!");
+        } else if (split.length == 1) {
+            throw new DukeArgumentException("You have not entered a task description!");
+        } else {}
```
If there is an empty block, is it better to comment on it to help others read your code?

Cp-John commented on src/main/java/duke/util/TaskStorage.java:
```diff
@@ -0,0 +1,116 @@
+     * @param tasks Existing user's tasks.
+     */
+    public void storeData(TaskList tasks) throws DukeStorageException {
+        try {
```
Perhaps using blank lines to separate different parts of your code is better?

Cp-John commented on src/main/java/duke/util/TaskStorage.java:
```diff
@@ -0,0 +1,116 @@
+                    LocalDateTime eventTime = LocalDateTime.parse(eventDetails, formatter);
+                    retrievedTasks.add(new Event(description, isDone, eventTime));
+                    break;
+                default:
```
You may put a break for the default block to facilitate reading.

Cp-John reviewed (commented):
The code is generally well-written and has good code quality.

zenlyj committed:
Adhere to coding standard

Made sure that Java Doc method headers begin with verb in plural form

sjq-sohjunqi commented on src/main/java/duke/TaskList.java:
```diff
@@ -6,44 +6,75 @@
+     * @param task The task to be added to the list of tasks.
      */
 
     public void addTask(Task task) {
```
The space here could be removed to help improve readability.

sjq-sohjunqi commented on src/main/java/duke/util/MessageFormatter.java:
```diff
@@ -6,25 +6,54 @@
+     *          current size of the list of tasks.
+     */
     public String formatAddCmdMsg(Task taskAdded, TaskList tasks) {
         return ("Got it. I've added this task:\n" + taskAdded +
```
Would you consider removing "Msg" from the name to make it shorter?

sjq-sohjunqi commented on src/main/java/duke/command/Command.java:
```diff
@@ -12,12 +12,17 @@
      */
-    public abstract String execute(TaskList tasks, Ui ui, TaskStorage storage);
+    public abstract String execute(TaskList tasks, MessageFormatter messageFormatter, TaskStorage storage);
 
```
Good use of abstract methods here!

sjq-sohjunqi commented:
> 
> 
> The code is generally well-written and has good code quality.

I agree! Overall, the code makes very good use of class inheritance and abstraction, making it very readable.

habi39 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    /**
+     * Executes the delete command.
```
Java documentation clearly defines the method and is very readable! Good job!

habi39 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,44 @@
+     * @param storage Updates tasks.txt of the new task.
+     * @return Duke's response after adding task to the list.
+     */
+    public String execute(TaskList tasks, MessageFormatter messageFormatter, TaskStorage storage) {
```
A clear and concise name for the method! Also a good JavaDoc header that describes this method!

habi39 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Class representing a Command.
+ */
+public abstract class Command {
```
According to Java coding standard, is it better to make it private here?

habi39 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,49 @@
+ * Class representing the Find Command.
+ */
+public class FindCommand extends Command {
+    private String keyword;
```
Adherence of the java Coding standards! Good Job!

habi39 commented on src/main/java/duke/gui/DialogBox.java:
```diff
@@ -0,0 +1,81 @@
+
+import java.io.IOException;
+import java.util.Collections;
+
```
Very good separation of imports here

habi39 commented on src/main/java/duke/util/MessageFormatter.java:
```diff
@@ -0,0 +1,91 @@
+     *          current size of the list of tasks.
+     */
+    public String formatAddCmdMsg(Task taskAdded, TaskList tasks) {
+        return ("Got it. I've added this task:\n" + taskAdded +
```
Would it be good to have a line break to improve readability?

habi39 reviewed (commented):
Overall, your code is easy to read and understand and is adhering to Java coding standard. There were only minor flaws that I could pick out but overall good job!

zenlyj committed:
Handle exception for invalid date/time syntax

When user enters an invalid date/time format

zenlyj committed:
Add assertions

zenlyj committed:
Add assertions for TaskStorage

zenlyj committed:
Improve code quality

* Reduce length of parse method by creating abstraction methods
* Remove use of "magic" values in parser by creating constants
* Resolve some inconsistencies in line breaks
* Shorten MessageFormatter method names
* Improve readability for string concatenations

zenlyj committed:
Implement Undo function

Allow user to undo Add, Delete, Done commands

zenlyj committed:
Merge pull request #2 from zenlyj/branch-A-CodeQuality

Improve code quality

zenlyj committed:
Merge pull request #3 from zenlyj/branch-C-Undo

Branch C-Undo

zenlyj committed:
Merge branch 'master' into branch-A-Assertions

zenlyj committed:
Merge pull request #1 from zenlyj/branch-A-Assertions

Branch A-Assertions

zenlyj committed:
Resolve errors from merging branch

zenlyj committed:
Fix minor error, and improve code quality

* Make assertion statement check for all task types
* Place break statement in default case
* Make file variable final

zenlyj committed:
Fix parsing error

Throws DukeException when user do not enter description for tasks

zenlyj committed:
Remove magic literals

Create constants to replace magic literals for task types and task
status

zenlyj committed:
GUI Improvements

* Change display picture of Duke and User
* Add padding between chat text and display picture
* Set title to windows

zenlyj committed:
Add more Java Docs

zenlyj committed:
Add User Guide

zenlyj committed:
Add Screenshot of application

zenlyj committed:
Improve Parser

Removes leading and trailing white spaces for task descriptions

zenlyj committed:
Handle parsing exception

Throw exception when user does not enter a description but tries to
enter a date and time

zenlyj committed:
Fix typo

Should be /at instead of at

zenlyj committed:
Styling fix

Place else block in correct position

zenlyj committed:
Add CheckStyle files

zenlyj committed:
Adhere to Coding Standards

Modified code to adhere to violations pointed out by Checkstyle
